src/core/annotations/qgsannotationlayer.cpp

```diff
@@ -225,7 +225,7 @@ bool QgsAnnotationLayer::isEmpty() const
   return mItems.empty();
 }
 
-QgsAnnotationItem *QgsAnnotationLayer::item( const QString &id )
+QgsAnnotationItem *QgsAnnotationLayer::item( const QString &id ) const
```

Comment:
shouldn't the return QgsAnnotationItem* type be const then ?

Reply:
That's open to discussion…

But we already have `QMap<QString, QgsAnnotationItem *> items() const { return mItems; }`, so either this one needs fixing (dropping const or returning const annotations) or it's ok to return non const member pointers in a const method.

Use case is that I'm using this method in a const method in a reimplementation of `QgsAnnotationLayer`.

Reply:
ok, was just asking in case this was an oversight. LGTM as such